### PR TITLE
[oatpp] upgrade to 1.1.0

### PR DESCRIPTION
**Pull request by Alberto-Izquierdo**
[oatpp] upgrade to 1.1.0

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Closes https://github.com/conan-io/conan-center-index/issues/1983

### CODE REVIEW

**Comment by uilianries** [comment]
@Alberto-Izquierdo is it ready for review? Please, update the PR state when ready.

File: recipes/oatpp/1.1.0/test_package/example.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#include "DeserializerTest.hpp"
```
Comment (uilianries) [inline recipes/oatpp/1.1.0/test_package/example.cpp:1]:
```suggestion
#include "DeserializerTest.hpp"
```

Please, use something smaller. An unique file, small, which validates the library lib with few method is enough.

**Comment by Alberto-Izquierdo** [comment]
> I don't think you need to add a new folder for 1.1.0, you can bump the config.yml and all/conandata.yml only.

@uilianries  The only problem is that there are some breaking changes between 1.0 and 1.1, how could we achieve using different test files with a single folder? Maybe some if/else in the CMakeLists.txt checking the version?

**Comment by uilianries** [comment]
@Alberto-Izquierdo what kind of changes? If it's an aggressive change and requires a new recipe, like options, new build flow, so I think splinting the folder is better, but for small changes, you can use `if tools.Version(self.version) >= "1.1.0"` and do what you you need.

**Comment by Alberto-Izquierdo** [comment]
@uilianries Sorry, I mean code changes, the code that worked before does not work anymore with the new version ([check this](https://github.com/conan-io/conan-center-index/pull/2024)) so I thought the only solution for this was creating a new folder with the changes in the examples.

**Comment by uilianries** [comment]
@Alberto-Izquierdo yes, if possible.

File: recipes/oatpp/1.1.0/test_package/example.cpp (hunk 1 above)
Comment (Alberto-Izquierdo) [reply recipes/oatpp/1.1.0/test_package/example.cpp]: Done

**Review by danimtb — APPROVED** [review]

**Pull request merged**